### PR TITLE
Added SQL Formatter tool

### PR DESCRIPTION
**Pull request by veler**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Internationalization and localization
- [ ] Other (please describe):

## What is the current behavior?

As a developer working with database, I'd like to be able to format some queries without having to paste them on a random website.

Issue Number: #136 

## What is the new behavior?

Added a SQL Formatter based on https://github.com/zeroturnaround/sql-formatter

It isn't perfect though. This is a literal translation of the JS library. Some optimizations can be made to reduce the allocation of strings.

## Other information

image

image

image

image

## Quality check

Before creating this PR, have you:

- [X] Followed the code style guideline as described in [CONTRIBUTING.md](https://github.com/veler/DevToys/blob/main/CONTRIBUTING.md)
- [ ] Verified that the change work in Release build configuration
- [X] Checked all unit tests pass

### CODE REVIEW

**Comment by veler** [comment]
@akarboush, please feel free to take a look at it :)

**Comment by akarboush** [comment]
@veler thanks for the mention.
I took a brief look at it, and It looks similar to mine since it's a direct translation. 
As you already mentioned, there's a lot of optimization to do (string allocation and some regex could be replaced with a plain C# methods too).

For the mean time, I'll try to optimize it on my branch and then propose the changes if there aren't any here yet

File: src/dev/impl/DevToys/Helpers/SqlFormatter/Core/Formatter.cs [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+        {
+            string formattedQuery = string.Empty;
+
+            Assumes.NotNull(_tokens, nameof(_tokens));
+            for (int i = 0; i < _tokens!.Count; i++)
+            {
+                _index = i;
+
+                Token token = TokenOverride(_tokens[i]);
+                if (token.Type == TokenType.LineComment)
```
Comment (Joaolfelicio) [inline src/dev/impl/DevToys/Helpers/SqlFormatter/Core/Formatter.cs:91]: We could clean a bit all the if else's by using pattern matching.

**Comment by veler** [comment]
> @veler thanks for the mention. I took a brief look at it, and It looks similar to mine since it's a direct translation. As you already mentioned, there's a lot of optimization to do (string allocation and some regex could be replaced with a plain C# methods too).
> 
> For the mean time, I'll try to optimize it on my branch and then propose the changes if there aren't any here yet

Sounds good. I think I will do a little bit of clean up before merging, but likely no optimization yet. While all these regex and strings are suboptimal, it seems to work relatively well as long as the SQL query isn't super long.

I'd suggest to do the optimizations in a separated PR, what do you think @akarboush ?

**Pull request merged**